### DISCUSSION TITLE
Index maintenance

### DISCUSSION THREAD

**Discussion by Akhi0587** [olahallengren/sql-server-maintenance-solution#557] [Questions & Answers]
hello,

The TimeLimit option doesn't work when it hit a lock with the index trying to maintain any solution for this?

TimeLimit
Set the time, in seconds, after which no commands are executed. By default, the time is not limited.

**Comment by jacobgexigo** [comment]
I assume you mean that it won't stop maintaining an index that it is already working on, correct?  If that is what you mean, then no there is no way around that in the solution.  The workaround would be to setup a separate job that runs at whatever time you need it to stop to see if there are any ALTER INDEX commands running and if so, kill that SPID.
